doc/dev-guide.rst

```diff
@@ -143,3 +134,37 @@ The module contains the following boiler-plate files:
   :external+pip:doc:`here <reference/build-system/pyproject-toml>`.
 
 .. _pip-compile: https://pip-tools.readthedocs.io/en/latest/#without-setup-py
+
+Preparing to raise a Pull Request
+---------------------------------
+
+Pre-commit compliance
+^^^^^^^^^^^^^^^^^^^^^
+
+Contributors who prefer to develop without pre-commit enabled will be required
+to ensure that any submissions pass all the checks described here before they
+can be accepted and merged.
+
+No judgement, we know pre-commit can be annoying if you're not used to it.
+This is in place in order to keep the code-base consistent so we can focus
+on the work at hand - rather than maintaining code readability and appearance.
+
+Module documentation updates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:mod:`katgpucbf` holds documentation within its code-base.
+:external+sphinx-apidoc:doc:`sphinx-apidoc <index>` provides a manner to generate
+module documentation as reStructuredText. If you, the developer, add or remove
+a module or file, execute the full ``sphinx-apidoc`` command below to regenerate
+the module documentation with your updates.
+
+.. code-block:: bash
+
+  user@host:~/katgpucbf$ sphinx-apidoc -efo doc/ src/
+
+.. note::
+
+    The above command will likely generate a :file:`modules.rst` file, which is
+    not necessary to commit. Furthermore, if you aren't comfortable generating
+    the documents straight away, you can append a ``--dry-run`` to the above
+    command to see what files will be generated.
```

Comment:
I was toying with splitting these two sentences into a `note::` and ` tip::`, but they're relatively short and looked strange when separated.

Reply:
I'd remove from "Furthermore" onwards. Git exists for a reason, so if you mess stuff up you can restore.